multichain-api-gateway/src/settings.rs

```diff
@@ -0,0 +1,87 @@
+use std::{net::SocketAddr, str::FromStr};
+
+use config::{Config, File};
+use serde::{de::IgnoredAny, Deserialize};
+use url::Url;
+
+/// An instance of the maintained networks in Blockscout.
+/// Semantic: (network, chain)
+/// e.g."blockscout.com/eth/mainnet" -> ("eth", "mainnet")
+#[derive(Deserialize, Clone, Debug, PartialEq, Eq)]
+pub struct Instance(pub String, pub String);
+
+/// Settings for the Blockscout API
+#[serde_with::serde_as]
+#[derive(Deserialize, Clone, Debug)]
+#[serde(default, deny_unknown_fields)]
+pub struct BlockscoutSettings {
+    /// The base URL of the Blockscout API.
+    pub base_url: Url,
+
+    pub instances: Vec<Instance>,
+
+    /// The number of concurrent requests to be made to the Blockscout API from a server's thread worker.
+    pub concurrent_requests: usize,
+
+    /// The timeout of waiting for response from the Blockscout API.
+    // Using chrono::Duration here instead of std::time::Duration,
+    // because there is no "Display" trait for std::time::Duration, hence trouble when calling "try_deserialize"
+    #[serde_as(as = "serde_with::DurationSeconds<i64>")]
+    pub request_timeout: chrono::Duration,
+}
+
+impl Default for BlockscoutSettings {
+    fn default() -> Self {
+        Self {
+            base_url: Url::parse("https://blockscout.com/").expect("should be correct base"),
+            instances: vec![],
```

Comment:
I suggest to add some instances by default as well (e.g. eth/mainnet and xdai/mainnet)